test/common/upstream/hds_test.cc

```diff
@@ -226,14 +226,18 @@ TEST_F(HdsTest, TestMinimalSendResponse) {
 
 TEST_F(HdsTest, TestStreamConnectionFailure) {
   EXPECT_CALL(*async_client_, start(_, _))
+      .WillOnce(Return(nullptr))
+      .WillOnce(Return(nullptr))
       .WillOnce(Return(nullptr))
       .WillOnce(Return(&async_stream_));
-  EXPECT_CALL(*retry_timer_, enableTimer(_));
+  EXPECT_CALL(*retry_timer_, enableTimer(_)).Times(3);
```

Comment:
You should be able to `EXPECT_CALL` on a specific time value here to demonstrate the jitter, e.g. `EXPECT_CALL(*retry_timer_, enableTimer(42));`. You will need to mock randomness for that to work.